lib/pdk/validators/metadata/metadata_syntax.rb

```diff
@@ -14,12 +14,39 @@ def self.pattern
         'metadata.json'
       end
 
-      def self.invoke(report, options = {})
+      def self.spinner_text(targets = [])
+        _('Checking metadata syntax (%{targets})') % {
+          targets: PDK::Util.targets_relative_to_pwd(targets).join(' '),
+        }
+      end
+
+      def self.create_spinner(threaded_spinner, targets = [], options = {})
+        PDK::Util.spinner_opts_for_platform(options)
+        @spinner = if options[:parallel]
+                     threaded_spinner.register("[:spinner] #{spinner_text(targets)}", options)
+                   else
+                     TTY::Spinner.new("[:spinner] #{spinner_text(targets)}", options)
+                   end
+        @spinner.auto_spin
+      end
+
+      def self.stop_serial_spinner
+        if @return_val.zero? && @spinner
+          @spinner.success
+        elsif @spinner
+          @spinner.error
+        end
+      end
+
+      def self.invoke(report, *args)
+        options = args.pop if args.last.is_a?(::Hash)
+        threaded_spinner = args.pop if args.last.is_a?(::TTY::Spinner::Multi)
         targets = parse_targets(options)
 
         return 0 if targets.empty?
 
-        return_val = 0
+        @return_val = 0
```

Comment:
Was there a reason this had to become an instance var? Something to do with threads?